include/stl2/detail/memory/uninitialized_copy.hpp

```diff
@@ -41,24 +40,38 @@ STL2_OPEN_NAMESPACE {
 		return {std::move(first), std::move(result)};
 	}
 
+	///////////////////////////////////////////////////////////////////////////
+	// uninitialized_copy [Extension]
+	//
 	template <InputRange Rng, __NoThrowForwardIterator O>
 	requires
-		Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>> &&
-		__ReferenceTo<O, value_type_t<O>>
+		Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>>
 	tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
 	uninitialized_copy(Rng&& rng, O result)
 	{
 		return __stl2::uninitialized_copy(
 			__stl2::begin(rng), __stl2::end(rng), result);
 	}
 
+	///////////////////////////////////////////////////////////////////////////
+	// uninitialized_copy [Extension]
+	//
+	template <InputRange IRng, __NoThrowForwardRange ORng>
+	requires
+		Constructible<value_type_t<iterator_t<ORng>>, reference_t<iterator_t<IRng>>>
+	tagged_pair<tag::in(safe_iterator_t<IRng>), tag::out(iterator_t<ORng>)>
+	uninitialized_copy(IRng&& irng, ORng& orng)
```

Comment:
It seems pretty safe to `uninitialized_copy` into an rvalue `span`.

Reply:
I can see benefits for lvalue `span` too.